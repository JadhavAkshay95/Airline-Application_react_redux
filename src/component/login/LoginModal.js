import Modal from 'react-bootstrap/Modal'
import GmailLogin from './GmailLogin'
import FbLogin from './FbLogin'
import Button from 'react-bootstrap/Button'
import React from 'react'
import PropTypes from 'prop-types'
import './login.scss'

const LoginModal = ({
  show,
  handleGoogleLogin,
  handleFbLogin,
  handleClose
}) => {
  return (
    <>
      <Modal
        show={show}
        size="lg"
        centered
        aria-labelledby="contained-modal-title-vcenter"
      >
        <Modal.Header>
          <Modal.Title>Login</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <div className="add-margin">
            <p>
              <GmailLogin handleLogin={handleGoogleLogin} />
            </p>
            <p className="fb-login">
              <FbLogin handleLogin={handleFbLogin} />
            </p>
          </div>
        </Modal.Body>
        <Modal.Footer>
          <Button variant="light" onClick={handleClose}>
            close
          </Button>
        </Modal.Footer>
      </Modal>
    </>
  )
}

export default LoginModal

LoginModal.propTypes = {
  show: PropTypes.bool,
  handleGoogleLogin: PropTypes.func,
  handleFbLogin: PropTypes.func,
  handleClose: PropTypes.func
}
