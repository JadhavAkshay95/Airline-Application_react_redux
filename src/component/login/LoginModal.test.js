import React from 'react'
import { shallow } from 'enzyme'
import LoginModal from './LoginModal'

const renderLoginModal = args => {
  const defaultProps = {
    show: true,
    handleGoogleLogin: () => {},
    handleFbLogin: () => {},
    handleClose: () => {}
  }
  const props = { ...defaultProps, ...args }
  return shallow(<LoginModal {...props} />)
}

describe('Render login modal component', () => {
  it('Render login modal', () => {
    let wrapper = renderLoginModal({ show: true })
    expect(wrapper).toMatchSnapshot()

    wrapper = renderLoginModal({ show: false })
    expect(wrapper).toMatchSnapshot()
  })
})
