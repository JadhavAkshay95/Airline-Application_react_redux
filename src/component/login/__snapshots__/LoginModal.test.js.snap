// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render login modal component Render login modal 1`] = `ShallowWrapper {}`;

exports[`Render login modal component Render login modal 2`] = `ShallowWrapper {}`;
